pydomcfg/accessor.py

```diff
@@ -43,12 +57,104 @@ def jpk(self) -> int:
 
     @jpk.setter
     def jpk(self, value: int):
-        if value <= 0:
-            raise ValueError("`jpk` MUST be > 0")
+        if value < 0:
+            raise ValueError("`jpk` MUST be >= 0 (use 0 to unset jpk)")
         self._jpk = value
 
+    @property
+    def nml_ref_path(self) -> str:
+        """
+        Path to reference namelist.
+
+        Returns
+        -------
+        str
+        """
+        return self._nml_ref_path
+
+    @nml_ref_path.setter
+    def nml_ref_path(self, value: str):
+        self._nml_ref_path = value
+
+    # domzgr methods
     @_jpk_check
     def zco(self, *args: Any, **kwargs: Any) -> Dataset:
         return Zco(self._obj, self._jpk)(*args, **kwargs)
 
     zco.__doc__ = Zco.__call__.__doc__
+
+    # Emulate NEMO DOMAINcfg tools
+    def from_namelist(self, nml_cfg_path_or_io: Union[str, Path, IO[str]]):
+        """
+        TODO
+        """
+
+        # Parse and return a ChainMap
+        nml = self._namelist_parser(nml_cfg_path_or_io)
+
+        # Pick zgr class
+        if nml["ln_zco"]:
+            zgr_class = Zco
+        else:
+            raise NotImplementedError("Only `Zco` has been implemented so far.")
+
+        # Inspect the __call__ signatures and get the approriate variables
+        # from the namelists. Replace 999999. with None
+        parameters = inspect.signature(zgr_class.__call__).parameters
+        kwargs = {
+            key: None if nml[key] == 999_999 else nml[key]
+            for key in parameters
+            if key != "self"
+        }
+
+        _check_namelist_entries(kwargs)
```

Comment:
I think we can now move `_check_namelist_entries` in `Zgr`, so we run the exact same checks to the namelists and the arguments of our classes. (I think we just have to make a small change to support optional arguments).

This is why I'm using `isinstance(value, int)` which is a less strict check compared to `type(value) == int`.
For example, `isinstance(bool, int)` is `True` which is OK for `pydomcfg` although NEMO would probably complain.
But I think it makes sense to use `isinstance` as we are not creating namelists to feed into NEMO.

Reply:
@malmans2 I guess this is where nml_meld will diverge then as  `1 != .true.` in NEMO speak. But I think that's fine.

Reply:
Right, if eventually pydomcfg will make use of nml_meld, I think we just need an extra argument list `isinstance=True` for pydomcfg but `False` by default.